tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -419,13 +419,25 @@ internal SpanContext CreateSpanContext(ISpanContext parent = null, string servic
 
                 if (traceContext == null)
                 {
-                    // If parent is SpanContext but its TraceContext is null, then it was extracted from
-                    // propagation headers. Create a new TraceContext (this will start a new trace) and initialize
+                    var propagatedTags = parentSpanContext.PropagatedTags;
+                    var samplingPriority = parentSpanContext.SamplingPriority;
+
+                    // When in appsec standalone mode, only distributed traces with the `_dd.p.appsec` tag
+                    // are propagated downstream, however we need 1 trace per minute sent to the backend, so
+                    // we unset sampling priority so the rate limiter decides.
```

Comment:
1) I answered the comment above 😄 

2) From the part 1 of ASM Standalone, does setting the sampling rule like this: https://github.com/DataDog/dd-trace-dotnet/blob/1adc8556ea523386338444ec36e94809bd55db57/tracer/src/Datadog.Trace/TracerManagerFactory.cs#L266-L271 isn't already forcing the rate limiter?

Reply:
Oh, interesting. Thanks for pointing that out. I had left [this comment](https://github.com/DataDog/dd-trace-dotnet/pull/5565#discussion_r1647561086) in the first PR #5565, and then when the comment was flagged as "outdated" by github, I thought this code was removed. But I see now it was just moved up a little, not removed 😅 

I'll copy the original comment here for reference:

> Have you tested that this sampler works the way you expect with this rate limiter and the global sampling rate? I think this will limit _all_ traces (APM and ASM traces) to 1 trace per minute. IIUC, according to the RFC you want:
> - keep _all_ ASM traces (not limited to 1 trace/min)
> - if there are no ASM traces, allow _at most_ 1 APM trace per minute to keep the service alive

But looking at the code again, it looks correct to me after all:
- You have a global sampling rate (100%), which means we _will_ use the rate limiter and not ignore it
- The sampling decision (which uses the rate limiter) only happens if a sampling decision is not set explicitly. Since ASM _is_ setting a sampling decision (user_keep/2) for all ASM traces (this is the part I forgot), then we _don't_ apply the rate limiter to those, only to the APM traces.

So... this looks good. Sorry for the extra trouble. I get nervous when I see sampling code 😂

Reply:
Oops yea sorry, I didn't saw your latest comment on the other PR.
Yes, this was tested end to end with the backend and it was working 😄.
Haha better be careful, no problem.